### ISSUE TITLE
MessengerTask not working

### ISSUE THREAD

**Issue by jogoool** [Guikingone/SchedulerBundle#218]
I just create a fresh Symfony project to test the bundle, I followed the documentation. the configured tasks are working fine but when I try to create MessengerTask it is not listed on the task list ` bin/console scheduler:list `.

| Q                | A
| ---------------- | ---
| PHP version?     | 7.4
| Bundle version?  | 0.7.1
| Symfony version? | 5.4

```yaml
scheduler_bundle:
    transport:
        dsn: 'filesystem://first_in_first_out'
    tasks:
        foo:
            type: 'command'
            command: 'debug:route'
            expression: '*/5 * * * *'
            description: 'A simple cache clear task'
            options:
                env: test
```

messenger config

``` yaml
framework:
    messenger:
        transports:
             async: '%env(MESSENGER_TRANSPORT_DSN)%'

        routing:
             'App\Message\Command\LogMessage': async
```

```php
namespace App\Task;

use App\Message\Command\LogMessage;
use SchedulerBundle\SchedulerInterface;
use SchedulerBundle\Task\MessengerTask;

class LoggerTask
{
    public function __invoke(SchedulerInterface $scheduler) {
        $task = (new MessengerTask('log-message', new LogMessage('log message')))
            ->setExpression('*/5 * * * *');
        $scheduler->schedule($task);
    }
}
```

output of  ` bin/console scheduler:list `.

```sql
+-------------+------+---------------------------+-------------+---------------------+---------------------------+-------------------------+-----------------------------+---------+------+
| Type        | Name | Description               | Expression  | Last execution date | Next execution date       | Last execution duration | Last execution memory usage | State   | Tags |
+-------------+------+---------------------------+-------------+---------------------+---------------------------+-------------------------+-----------------------------+---------+------+
| CommandTask | foo  | A simple cache clear task | */5 * * * * | Not executed        | 2021-12-23T19:40:00+00:00 | Not tracked             | Not tracked                 | enabled |      |
+-------------+------+---------------------------+-------------+---------------------+---------------------------+-------------------------+-----------------------------+---------+------+
```

**Comment by Guikingone** [comment]
Hi @jogoool 👋🏻 

Hum, this seems strange as the task is not marked as "queued", sadly, it's an edge case which is not tested 😅 

I'm gonna investigate it thanks to the informations you gave, could you specify which PHP you use? 

Thanks 🙂

**Comment by Guikingone** [comment]
Just a small question, I see the class `LoggerTask`, could you be more specific on how this class is used? 

Is it a service?

**Comment by jogoool** [comment]
Hello @Guikingone 

i try to set the task queued to true but I get the same result.

```php
        $task = (new MessengerTask('log-message', new LogMessage('this is the message')))
            ->setExpression('*/5 * * * *')
            ->setQueued(true);
```

the php version is 7.4 and the autowire of symfony is enabled, so LoggerTask is defined as a service.

**Comment by Guikingone** [comment]
Hi @jogoool 👋🏻 

Sorry for the delay, happy new year BTW 😄 

Thanks for the extra information, I'm gonna investigate this behaviour. 

After looking and adding new tests locally, the `MessengerTask` can be scheduled without issues (it appears in the related command), the issue you're facing can be related to the DIC and how `LoggerTask` is registered into it, I'm gonna see if the bundle can do something about this.

**Comment by Guikingone** [comment]
Hi @jogoool 👋🏻 

Sorry for the delay, so, here's the actual situation regarding this issue:

- The issue you're facing is not a "direct" issue, that's a feature lack as the container CANNOT trigger a schedule directly via `__invoke()`, there's no factories or "method call" (as Symfony calls it) to handle this situation.

- If I want to support this type of "trigger", I cannot add it in `0.7` as it's a BC breaks / new feature so it will be added in `0.9` (which require PHP `8.0`).

Is there any issue regarding this feedback or am I lacking any usage that you can face regarding this issue? Let me know 🙂

**Comment by Guikingone** [comment]
Hi @jogoool 👋🏻 

A fix will be released via `0.9.0` soon, thanks for the issue 🙂
